packages/gatsby/src/redux/types.ts

```diff
@@ -149,7 +149,7 @@ export interface IGatsbyState {
     PLUGINS_HASH: Identifier
   }
   componentDataDependencies: {
-    nodes: Map<string, Set<string>>
+    nodes: Map<string | number, Set<string>>
```

Comment:
This should be just `string` (keys should be node ids which are strings). 

What caused the need for ` | number`? If there is any code that use numbers we need to fix it there.

This also potentially cause some unnecessary casting happening right now in this PR

Reply:
In the test file, some nodes are mocked as `number`, I'll go ahead and fix there too.

Reply:
I'll go ahead and commit a change because you adjusted tests but seem to missed this one :) I already verified locally that typecheck is happy after dropping ` | number` here so I should update it in few minutes